### PR TITLE
Added fix for FHIR DocumentReference with attachment with hash field

### PR DESCRIPTION
**Pull request by ccorley**
Updated the connect json encoder to handle the FHIR type Base64Binary

### CODE REVIEW

File: connect/routes/fhir.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ async def post_fhir_data(
         raise HTTPException(status_code=404, detail=f"/{resource_type} not found")
 
     if resource_type != request_data.get("resourceType"):
-        msg = f"request {request_data.get('resource_type')} does not match /{resource_type}"
+        msg = f"request {request_data.get('resourceType')} does not match /{resource_type}"
```
Comment (ccorley) [inline connect/routes/fhir.py:76]: Minor unrelated fix: request_data.get('resource_type')} is 'None'

File: connect/support/encoding.py [hunk 2]
```diff
@@ -28,6 +28,8 @@ def default(self, o: Any) -> Any:
             return o.isoformat()
         elif isinstance(o, uuid.UUID):
             return str(o)
+        elif isinstance(o, bytes):
+            return base64.b64encode(o).decode()
```
Comment (ccorley) [inline connect/support/encoding.py:32]: Needed to support fhir.resources fhirtypes.Base64Binary type.

**Review by dixonwhitmire — APPROVED** [review]
